### ISSUE TITLE
Using other AI model Ollama

### ISSUE THREAD

**Issue by AlazaziAmr** [vintasoftware/django-ai-assistant#184]
While using Ollama (with llama) the outputs/responses are not good and accurate as openai, is there a special implementation to use Ollama?

**Comment by fjsj** [comment]
Hi @AlazaziAmr I believe that's the model fault, not anything specific caused by django-ai-assistant.
But you can double-check what's the input being passed to the model by using Langsmith: https://docs.smith.langchain.com/
All you have to do is set the env vars:
```
LANGCHAIN_TRACING_V2=true
LANGCHAIN_API_KEY=<your-api-key>
```
If you're using the example projects, set those at your .env file.

**Comment by AlazaziAmr** [comment]
Hi @fjsj, everything is set correctly, but the Ollama(with llama) can't find the specific function to do the work as the OpenAI.
Maybe it's as you said "the model fault".

**Comment by fjsj** [comment]
the specific ollama model you're using is really fine-tuned to perform function calling? @AlazaziAmr

**Comment by AlazaziAmr** [comment]
As mentioned in their website it supports tool calling. @fjsj 
![ex](https://github.com/user-attachments/assets/5b0e279a-aecc-4140-b606-a8389d00a7cb)
same example showen in the above picture performed with gpt-4o and work perfectly.

**Comment by AlazaziAmr** [comment]
No the function tools not passed

**Comment by fjsj** [comment]
What ChatModel are you using? Should be one that supports "Tool calling" from this table: https://python.langchain.com/docs/integrations/chat/#featured-providers

Is it really ChatOllama?

**Comment by AlazaziAmr** [comment]
Yes it's ChatOllama, and I think the function tools are passed to acording to the this picture 
![langsmith](https://github.com/user-attachments/assets/63b1c5cf-7d5b-411f-8152-00c75aed7d2e)
but maybe the model does not know it !?

**Comment by AlazaziAmr** [comment]
Yes it's ChatOllama, and I think the function tools are passed to acording to the this picture 
![langsmith](https://github.com/user-attachments/assets/63b1c5cf-7d5b-411f-8152-00c75aed7d2e)
but maybe the model does not know it !?

**Comment by AlazaziAmr** [comment]
Yes it's ChatOllama, and I think the function tools are passed to acording to the this picture 
![langsmith](https://github.com/user-attachments/assets/63b1c5cf-7d5b-411f-8152-00c75aed7d2e)
but maybe the model does not know it !?

**Comment by fjsj** [comment]
Yes, it seems right. Looks like it's a model problem.

**Comment by AlazaziAmr** [comment]
Yes, I'll try to dig deeper into it, and find the problem, and let you know about the updates.

Thank you so much for your time, appreciate it.

**Comment by heikojan2010** [comment]
I've submitted Pull Request #185 which uses local (o)Llama3.1 for the weather example app.
